machinist/mserver/controller.go

```diff
@@ -1,16 +1,15 @@
-package controller
+package mserver
 
 import (
 	"github.com/enfabrica/enkit/machinist/rpc/machinist"
 	"log"
 	"sync"
 )
 
-type Worker struct {
-}
-
 type Controller struct {
-	workers sync.Map
+	workers        sync.Map
+	ConnectedNodes map[string][]string
```

Comment:
A couple notes here:
- `workers` was meant to be your `ConnectedNodes` and `sync.RWMutex` together? a `sync.Map` is pretty much an uglier map, with built in locking. You may want to revert the ConnectedNodes and addition of the lock?
- `ConnectedNodes` ... would probably want it to be private? so lowercase. Especially given that it requires a lock to be accessed, so someone with access to Controller should only read it through accessors.
- Rather than `map[string][]string`, I recommend `map[string]*Node` (eg, a struct that records the tags, when the last ping was received, the ip address of the node, ... etc, basis for the web page showing the list of nodes, for example)

Reply:
I don't want to use sync.Map until generics hit in 2 months or so just because I don't want to be typecasting everything.

You're right about the private access with a getter and the storing a struct, this was just for prototyping sake. Changed

Reply:
*Changed the connected node accessors and naming. Also, concurrent reading on maps is supported.